### PR TITLE
Update `state_transition` rule in Text2Model

### LINKED ISSUE #231: Raise `DetectionError` instead of creating wrong model
### Discussed in https://github.com/biomass-dev/biomass/discussions/223

Prepare `test_ubiquitination.txt` including:
```
A ubiquitinates B --> uB
```

Since there is no rule like 'ubiquitinate', we expect
```
# This is what I expect.
DetectionError: Unregistered words in line1: A ubiquitinates B --> uB
```

However, when you run the following code:
```python
testmodel = Text2Model("test_ubiquitination.txt")
testmodel.convert(overwrite=True)
```

This does NOT raise errors but creates model.
When you look into the resulting model, you will soon find the resulting model is wrong.
For example, in `name2idx/species.py`:
```python
from dataclasses import make_dataclass
from typing import Dict, List

NAMES: List[str] = [
    "AubiquitinatesB", # <-- this is not desired :(
    "uB",
]

...
```

It seems that this unregistered word is recognized as [`state transition rule`](https://github.com/biomass-dev/biomass/pull/169).

### PR DESCRIPTION
**Pull request by himoto**
- Update `state_transition` rule based on the discussion #223
- Add test for the updated rule
- Close #231 when merged

### CODE REVIEW

**Review by formersbach — COMMENTED** [review]
Hey Hiroaki!
What's the purpose of the .strip() here? 
Other than that all looks good.

File: biomass/construction/reaction_rules.py [hunk 1]
```diff
@@ -1711,12 +1711,12 @@ def state_transition(self, line_num: int, line: str) -> None:
                 two_species = description[0].split(arrow)
                 reactants = (
                     two_species[0].replace(" ", "").split("+")
-                    if len(two_species[0].split(" ")) > 1 and "+" not in two_species[0]
+                    if len(two_species[0].strip().split(" ")) > 1 and "+" not in two_species[0]
```
Comment (formersbach) [inline biomass/construction/reaction_rules.py:1714]: This .strip() I mean.

Comment (himoto) [reply biomass/construction/reaction_rules.py:1714]: Thanks for your review!
If I remember right there were some test cases that failed without this `strip()` but not sure now.
I will create a new commit without it.
If there's no problem without this `strip()`, let's go without it.

Comment (himoto) [reply biomass/construction/reaction_rules.py]: Hi @formersbach,
As you can see in the latest commit, the `strip()` that you mentioned was not necessary. Therefore I would like to go without it as you originally suggested.
Thanks again for your comment!

**Pull request merged**